framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -484,10 +485,16 @@ public Boolean visitDeclared_Declared(
   /**
    * A helper class for visitDeclared_Declared. There are subtypes of DefaultTypeHierarchy that need
    * to customize the handling of type arguments. This method provides a convenient extension point.
+   *
+   * @param subtype a possible subtype
+   * @param supertype a possible supertype
+   * @param subtypeRaw whether {@code subtype} is a raw type
+   * @param supertypeRaw whether {@code supertype} is a raw type
+   * @return the result of visiting type args
    */
   protected boolean visitTypeArgs(
-      final AnnotatedDeclaredType subtype,
-      final AnnotatedDeclaredType supertype,
+      AnnotatedDeclaredType subtype,
+      AnnotatedDeclaredType supertype,
```

Comment:
Below there is also
```java
    if (subtypeTypeArgs.isEmpty()) {
      return true;
    }
```
How can that happen?

Reply:
If subtype doesn't have any type arguments.  This method is called even if `subtype` doesn't have type arguments.

Reply:
A reader might expect that if `subtype` has no type arguments,  `subtypeRaw` would be `true`.  Is that assumption incorrect?  Could you clarify?

Reply:
`subtype` could be `String` which is neither raw nor does it have type argumetns.